File: course-schedule-iii.cc
// Course Schedule III My SubmissionsBack To Contest
class Solution {
public:
  int scheduleCourse(vector<vector<int>>& as) {
    sort(as.begin(), as.end(), [](const vector<int>& x, const vector<int>& y) {
      return x[1] < y[1];
    });
    priority_queue<int> q;
    int s = 0;
    for (auto &a: as) {
      q.push(a[0]);
      if ((s += a[0]) > a[1])
        s -= q.top(), q.pop();
    }
    return q.size();
  }
};

File: course-schedule-iii.rs
// Course Schedule III
impl Solution {
    pub fn schedule_course(mut courses: Vec<Vec<i32>>) -> i32 {
        use std::collections::BinaryHeap;
        courses.sort_unstable_by_key(|x| x[1]);
        let mut h = BinaryHeap::new();
        let mut s = 0;
        for c in courses.into_iter() {
            s += c[0];
            h.push(c[0]);
            if s > c[1] {
                s -= h.pop().unwrap();
            }
        }
        h.len() as i32
    }
}
